clickstream-event-visualiser-ui/src/main/java/clickstream/eventvisualiser/ui/internal/data/datasource/CSEvDatasourceImpl.kt

```diff
@@ -25,13 +32,16 @@ internal class CSEvDatasourceImpl private constructor(private val evObserver: CS
         evObserver.removeObserver(eventCallback)
     }
 
-    override fun startObserving() {
-        coroutineScope = CoroutineScope(SupervisorJob())
-        evObserver.addObserver(eventCallback)
+    override suspend fun addEvents(eventList: List<CSEvEvent>) {
+
     }
 
-    override suspend fun addEvents(eventList: List<CSEvEvent>) {
+    override val connectionFlow: Flow<Boolean>
+        get() = internalConnectionFlow.asStateFlow()
 
+    override fun startObserving() {
+        coroutineScope = CoroutineScope(SupervisorJob())
```

Comment:
should we cancel this scope in the stopObserving?

Reply:
We are cancelling it in `stopObserving` [here](https://github.com/gojek/clickstream-android/blob/ee102fb7091d08681205658f7fccdd4b587055e8/clickstream-event-visualiser-ui/src/main/java/clickstream/eventvisualiser/ui/internal/data/datasource/CSEvDatasourceImpl.kt#L31).